### PR TITLE
Multiple fixes for tests on Windows

### PR DESCRIPTION
**Pull request by carlescufi**

### CODE REVIEW

File: src/west/manifest.py [hunk 1]
```diff
@@ -278,7 +278,7 @@ def __init__(self, name, remote, defaults, path=None, clone_depth=None,
         self.remote = remote
         self.url = remote.url + '/' + name
         self.path = path or name
-        self.abspath = os.path.join(util.west_topdir(), self.path)
+        self.abspath = os.path.join(util.west_topdir(), os.path.normpath(self.path))
```
Comment (mbolivar) [inline src/west/manifest.py]: Is this right? Why isn't it `os.path.normpath(os.path.join(util.west_topdir(), self.path))`?

File: tests/west/manifest/test_manifest.py [hunk 2]
```diff
@@ -61,8 +61,9 @@ def test_no_defaults():
     # Check the projects are as expected.
     for p, e in zip(manifest.projects, expected):
         deep_eq_check(p, e)
-    assert all(p.abspath == '/west_top/' + p.path for p in manifest.projects)
-
+    assert all(os.path.normpath(p.abspath) ==
```
Comment (mbolivar) [inline tests/west/manifest/test_manifest.py]: I thought we guaranteed that p.abspath is a normpath in the previous patch?

File: tests/west/manifest/test_manifest.py [hunk 3]
```diff
@@ -107,8 +108,9 @@ def test_default_clone_depth():
     # Check that the projects are as expected.
     for p, e in zip(manifest.projects, expected):
         deep_eq_check(p, e)
-    assert all(p.abspath == '/west_top/' + p.path for p in manifest.projects)
-
+    assert all(os.path.normpath(p.abspath) ==
```
Comment (mbolivar) [inline tests/west/manifest/test_manifest.py]: Same comments apply here

File: src/west/manifest.py (hunk 1 above)
Comment (carlescufi) [reply src/west/manifest.py]: Will fix

**Comment by carlescufi** [comment]
@mbolivar comments addressed, thanks for the review.

File: tests/west/project/test_project.py [hunk 4]
```diff
@@ -39,7 +40,10 @@ def cmd(cmd):
 
     # cmd() takes the command as a string, which is less clunky to work with.
     # Split it according to shell rules.
-    split_cmd = shlex.split(cmd)
+    if sys.platform == 'win32':
+        split_cmd = shlex.split(cmd, posix=False)
```
Comment (ulfalizer) [inline tests/west/project/test_project.py]: A comment explaining why this is here would be nice:

```python
# posix=False prevents backslashes from getting stripped from paths on Windows
```

Wonder if we could just pass `posix=False` for both. `shlex.split()` it works the same regardless of platform (checked the code), and we never pass backslashes other than in the manifest path.

Comment (ulfalizer) [reply tests/west/project/test_project.py]: Wait... there's a simpler solution here:

    cmd += ' -m "{}"'.format(manifest_path)

Quoting the path before splitting prevents the backslashes from getting removed.

Comment (carlescufi) [reply tests/west/project/test_project.py]: OK, nice suggestion. I will go for that one.

Comment (carlescufi) [reply tests/west/project/test_project.py]: @ulfalizer done

**Comment by ulfalizer** [comment]
Maybe the commit message could be updated too:

    Quote the path to the manifest before splitting so that
    backslash path separators don't get removed on
    Windows.

**Comment by carlescufi** [comment]
@ulfalizer updated the commit message

**Pull request merged**